doc/03_research/03_planning/04_decision_making.md

```diff
@@ -0,0 +1,256 @@
+
+#### Disadvantages
+
+* multiple decision trees necessary
+* prediction independent of previous state
+
+## Previous approaches
+
+### PAF21-1
+
+#### State machine
+
+* 2 state machines: one for maneuvers, one for speed control
+* Speed control more complex, when to brake seems like the most challenging task
+
+#### Take away
+
+* Some states seem to be comparable to what we are required to accomplish by the leaderboard
+* Our task might be more complex, needs additional states and transitions
+* I'm uncertain about an extra speed state, might be easier to handle that more locally by the local planner, maybe in combination with an observer element that keeps track of the surrounding by processing the information from `Perception`
+
+### PAF21-2
+
+#### No clear concept
+
+* some sort of state machine integrated in local planner
+* obstacle planner for dynamic obstacles (pedestrians, cars, bicycles)
+* useful parameters which we could adapt
+* path prediction for obstacles
+* obstacles are only interesting if they cross the path of the ego vehicle
+
+#### Take away
+
+* Obstacle planner might be useful for dynamic obstacle detection if not handled elsewhere
+* path prediction might reduce the number objects tracked that we could interfere with
+* Also, if we adapt our local plan this path prediction of other vehicles might come in handy
+* On the other hand, overhead to keep track of vehicles and maybe repredict paths if some vehicles change direction
+
+### PSAF1(2020)
+
+#### State machine
+
+* Three driving functions: Driving, stopping at traffic light, stopping at stop sign
+* First project iteration so state machine more simple
+* still covers many important scenarios
+
+#### Take away
+
+* Good starting point to have a minimal viable state machine
+* Need adaption depending on what information we are getting forwarded/process in the planning module
+
+### PSAF2(2020)
+
+#### Decision tree
+
+* This team used a decision tree to cover the major driving scenarios
+* Within the scenarios the actions are more linear
+* Reminds me of the execution of a state where driving scenarios are the states and the execution the things our local planner should do within that state
+
+#### Take Away
+
+* Even though the approach is different, the execution might be similar to the other team algorithms
+* We might not be interested in a decision tree as we want to keep the option to switch to a Markov chain, which would be more overhead if we start with a decision tree
+
+## Python or ROS libraries for these decision-making algorithms
+
+### State machines
+
+#### SMACH
+
+* Task-level architecture for creating state machines for robot behaviour.
+* Based on Python
+* Fast prototyping: Quickly create state machines
+* Complex state machines can easily be created
+* Introspection: smach_viewer provides a visual aid to follow the state machine executing its tasks
+  * smach_viewer is unmaintained and does not work with noetic
```

Comment:
Wie problematisch ist das? Können wir das zum laufen bringen oder kommt SMACH daher eigl nicht in Frage?

Reply:
Also, es gab Teams in den letzten Jahren, die Entscheidungsbäume verwendet haben und die haben diese Eigenschaft auch nicht erfüllt. Ich denke die einzige Stelle wo es eventuell problematisch werden könnte, ist die Notbremsung. Ansonsten würde ich sagen, wir werden es nur rausfinden, wenn wir anfangen das zu implementieren